### PR TITLE
Fixes #1175 Cleaned up the use of logging NullHandler

### LINKED ISSUE #1175: Add NullHandler back in
PR #1172 has removed any NullHandlers from any pywbem loggers.

After researching best practices, the should be added back in.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
Ready for review and merge.

### CODE REVIEW

File: pywbem/_logging.py [hunk 1]
```diff
@@ -48,8 +48,25 @@
...
-The pywbem loggers log at the :attr:`py:logging.DEBUG` logging level.
+Pywbem uses the :attr:`py:logging.DEBUG` logging level for both loggers.
+
+The best practices for Python libraries suggest that libraries such as pywbem
+have a null handler attached to the Python logger for the package (the
+logger named `'pywbem'`, in this case), because since Python 2.7, the Python
+root logger will by default (i.e. when not being configured) print log
+records of logging level :attr:`py:logging.WARNING` or greater to `sys.stderr`.
+Having a null handler on the logger for the library prevent such log events
+to propagate up to the Python root logger, and thus the library will be silent
```
Comment (KSchopmeyer) [inline pywbem/_logging.py]: Change "log events to propagate" to "log events from propagating". Don't you just love english..

You can also remove the word thus in "thus the library will be slient"

File: pywbem/_listener.py [hunk 2]
```diff
@@ -762,31 +781,14 @@ def logger(self):
...
 
-        Each listener object has its own separate logger object that is
-        created via :func:`py:logging.getLogger`.
+        Each listener object has its own separate logger object with the name:
 
-        The name of this logger object is:
+          `'pywbem.listener.{id}'`
 
-          ``pywbem.listener.{id}``
+        where `{id}` is a unique string for each listener object.
```
Comment (KSchopmeyer) [inline pywbem/_listener.py]: We should probably define the scope of unique. Process, global, etc.

File: pywbem/_listener.py [hunk 3]
```diff
@@ -762,31 +781,14 @@ def logger(self):
...
         :class:`py:logging.Logger`: Logger object for this listener.
 
-        Each listener object has its own separate logger object that is
-        created via :func:`py:logging.getLogger`.
+        Each listener object has its own separate logger object with the name:
 
-        The name of this logger object is:
+          `'pywbem.listener.{id}'`
 
-          ``pywbem.listener.{id}``
```
Comment (KSchopmeyer) [inline pywbem/_listener.py]: Did not mean to start a conversation just single review comments

File: pywbem/__init__.py [hunk 4]
```diff
@@ -50,6 +51,17 @@
 
 from ._version import __version__  # noqa: F401
 
+# TODO: Verify whether pywbem needs a logging null handler.
+# import logging
+# try:  # Python 2.7+
```
Comment (KSchopmeyer) [inline pywbem/__init__.py]: Change comment to # Python 2.7+ include NullHandler

**Comment by KSchopmeyer** [comment]
I want to rebase this for the other test and run that before I finish but the code comments are in

**Comment by KSchopmeyer** [comment]
Forget comment about rebase.  That was not needed. The section on best practices really helps.

Only one other comment. Should we mention in _logging.py that logging for indicaitons is in another section?

File: pywbem/_logging.py (hunk 1 above)
Comment (andy-maier) [reply pywbem/_logging.py]: DONE, both.

File: pywbem/_listener.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/_listener.py]: DONE, changed to: "a string that is unique for each :class:`~pywbem.WBEMListener`
object within the Python process."

File: pywbem/_listener.py (hunk 3 above)
Comment (andy-maier) [reply pywbem/_listener.py]: Ok

File: pywbem/__init__.py (hunk 4 above)
Comment (andy-maier) [reply pywbem/__init__.py]: DONE

**Comment by andy-maier** [comment]
The latest commit resolves all comments as proposed. Considering this PR approved.

**Pull request merged**